plugins/user_mapping/src/user_claim.cpp

```diff
@@ -0,0 +1,64 @@
+#include "interface.h"
+#include "user_claim.hpp"
+
+#include <cstdlib>
+#include <cstring>
+#include <exception>
+#include <string>
+
+#include <spdlog/spdlog.h>
+
+namespace
+{
+	std::string claim_to_match;
+} // anonymous namespace
+
+auto init(const nlohmann::json& _config) -> void {
+	const auto claim{_config.find("irods_user_claim")};
```

Comment:
so... this is hardcoded here?    or maybe i'm missing a level of indirection...

Reply:
We need to look for something in the configuration... This is no more hard-coded than `local_file.cpp`'s `file_path`.

We use the value that `irods_user_clam` is given to then get the iRODS username that the claim specified by `irods_user_claim` has in an authenticated OpenID user.

Very briefly, `irods_user_claim`  (gives) -> `claim_name` (contains) -> `real_irods_username`.

Reply:
Resolving!